lua/lvim/lsp/init.lua

```diff
@@ -35,11 +40,6 @@ function M.common_capabilities()
     },
   }
 
-  local status_ok, cmp_nvim_lsp = pcall(require, "cmp_nvim_lsp")
```

Comment:
the deprecation message is a bit confusing, looking at the code this should work

```lua
  local capabilities = vim.lsp.protocol.make_client_capabilities()
  capabilities.textDocument.completion.completionItem.snippetSupport = true
  local status_ok, cmp_nvim_lsp = pcall(require, "cmp_nvim_lsp")
  if status_ok then
    capabilities = cmp_nvim_lsp.default_capabilities(capabilities)
  end
```

Reply:
@kylo252  if we have nvim-cmp-lsp, then we don't need to call `vim.lsp.protocol.make_client_capabilities`

if we don't have it, then we should call and do the overrides

Reply:
but this call is only using adding `.textDocument.completion`, see: https://github.com/hrsh7th/cmp-nvim-lsp/blob/389f06d3101fb412db64cb49ca4f22a67882e469/lua/cmp_nvim_lsp/init.lua#L24

Reply:
oh that makes sense, let me fix it

Reply:
actually this should work since lspconfig has gotten updated as well, see https://github.com/hrsh7th/cmp-nvim-lsp/commit/389f06d3101fb412db64cb49ca4f22a67882e469 and https://github.com/neovim/nvim-lspconfig/commit/b6d9e427c9fafca5b84a1f429bef4df3ef63244b